src/launchbox/token/LaunchboxERC20.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts/proxy/Clones.sol";
+import {LaunchboxExchange} from "../exchange/LaunchboxExchange.sol";
+
+contract LaunchboxERC20 is ERC20Upgradeable {
+    string public metadataURI;
+    address payable public launchboxExchange;
+
+    error MetadataEmpty();
+    error CannotSellZeroTokens();
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(string memory _name, string memory _symbol, uint256 _maxSupply, uint256 _marketCapThreshold, string memory _metadataURI, address _launchboxExchangeImplementation, address  _uniswapRouter) external initializer returns(address) {
+        __ERC20_init(_name, _symbol);
+        if(bytes(_metadataURI).length == 0) {
+            revert MetadataEmpty();
+        }
+        if(_maxSupply == 0) {
+            revert CannotSellZeroTokens();
+        }
+
+        metadataURI = _metadataURI;
+
+        launchboxExchange = payable(Clones.clone(_launchboxExchangeImplementation));
+        _mint(launchboxExchange, _maxSupply);
```

Comment:
Ideally, we would want to calculate the fees and pass that into the `intitalize` method. Fee should be set by admin in the factory contract. Fee will be a % of the token supply that will be sent to a Launchbox managed address. see here: https://github.com/metastable-labs/contracts/blob/launchbox-erc20/src/launchbox/token/LaunchboxERC20Factory.sol#L69-L71

Reply:
<img width="798" alt="Screen Shot 2024-06-13 at 7 40 42 AM" src="https://github.com/metastable-labs/contracts/assets/38771353/a9d6e1e6-94e8-4daa-a2a9-ba9fce60b2c9">
how I'm thinking about distributing the supply